### PR TITLE
Fix Bazel build and add a GitHub action to exercise the build on different platforms

### PR DESCRIPTION
**Pull request by KimuraYu45z**
This is an extended branch from https://github.com/KimuraYu45z/protobuf-rules-gen/tree/master
so plz see https://github.com/FirebaseExtended/protobuf-rules-gen/pull/42 at first.

In this branch, I wrote the github actions workflow to emit the built binary files across platform matrix.
This can help M1 mac users and so on.

### CODE REVIEW

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
     if "com_google_protobuf" not in native.existing_rules():
-        native.http_archive(
+        http_archive(
+            name = "rules_python",
+            url = "https://github.com/bazelbuild/rules_python/releases/download/0.3.0/rules_python-0.3.0.tar.gz",
+            sha256 = "934c9ceb552e84577b0faf1e5a2f0450314985b4d8712b2b70717dc679fdc01b",
+        )
+        http_archive(
+            name = "bazel_skylib",
+            urls = [
```
[the remark this answers is not in the document]
Comment (KimuraYu45z) [reply bazel/repositories.bzl:13]: I fixed.

File: bazel/repositories.bzl [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
-protobuf_sha256 = "c0ab1b088e220c1d56446f34001f0178e590270efdef1c46a77da4b9faa9d7b0"
-
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
 def protobuf_rules_gen_repositories():
     if "com_google_protobuf" not in native.existing_rules():
-        native.http_archive(
+        http_archive(
+            name = "rules_python",
+            url = "https://github.com/bazelbuild/rules_python/releases/download/0.3.0/rules_python-0.3.0.tar.gz",
```
[the remark this answers is not in the document]
Comment (KimuraYu45z) [reply bazel/repositories.bzl]: I fixed.

File: bazel/repositories.bzl [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+            ],
+        )
+        http_archive(
             name = "com_google_protobuf",
-            sha256 = protobuf_sha256,
-            strip_prefix = "protobuf-" + protobuf_commit,
-            url = "https://github.com/google/protobuf/archive/" + protobuf_commit + ".tar.gz",
+            sha256 = "528927e398f4e290001886894dac17c5c6a2e5548f3fb68004cfb01af901b53a",
+            strip_prefix = "protobuf-3.17.3",
+            urls = ["https://github.com/google/protobuf/archive/v3.17.3.zip"],
```
[the remark this answers is not in the document]
Comment (KimuraYu45z) [reply bazel/repositories.bzl]: I fixed.

File: .github/workflows/binary.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        arch: [amd64, arm64]
+        targetos: [windows, darwin, linux]
+    name: ${{ matrix.arch }}-${{ matrix.targetos }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          sudo apt install apt-transport-https curl gnupg
+          curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg
+          sudo mv bazel.gpg /etc/apt/trusted.gpg.d/
+          echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
```
[the remark this answers is not in the document]
Comment (KimuraYu45z) [reply .github/workflows/binary.yml]: https://docs.bazel.build/versions/main/install-ubuntu.html
It was copied from official document.
Both can work.

I followed you and fixed.

**Review by var-const — APPROVED** [review]
Thank you for the fix!

My only remaining comment is to expand the PR description to be a little more descriptive. I went ahead and renamed it for faster turnaround -- please let me know if you'd prefer any changes (the PR name can be changed even when it's merged).

**Pull request merged**